satellite/audit/reservoir.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"math/rand"
+	"storj.io/storj/pkg/pb"
+)
+
+// Reservoir holds a certain number of segments to reflect a random sample
+type Reservoir struct {
+	Segments []*pb.RemoteSegment
+	IsVetted bool
+	NumSlots int
+}
+
+type reservoirConfig struct {
+	numVettedSlots   int
+	numUnvettedSlots int
+}
+
+// NewReservoir instantiates a Reservoir
+func NewReservoir(isVetted bool, config reservoirConfig) *Reservoir {
+	var numSlots int
+	if isVetted {
```

Comment:
imo, we should take the number of slots in as an argument to `NewReservoir`, and the config for vetted vs. unvetted should be in the observer logic. That way this can be a generic reservoir sampler that doesn't have to be specific to audits

Reply:
If we want to be truly generic, then instead of `Segments` field we should have a more generic field `Slots` of type `interface{}`.

Reply:
it's actually a little convoluted to make the struct accept the slice of segments as a field (https://golang.org/doc/faq#convert_slice_of_interface) and later append to it, so I think I'm just going to leave it as []*pb.RemoteSegment for now.